src/vs/workbench/contrib/extensions/browser/extensionsViews.ts

```diff
@@ -649,6 +655,28 @@ export class ExtensionsListView extends ViewletPanel {
 			});
 	}
 
+	// {{SQL CARBON EDIT}}
+	private getVisualizerExtensions(token: CancellationToken): Promise<IPagedModel<IExtension>> {
+		return this.extensionsWorkbenchService.queryLocal()
+			.then(result => result.filter(e => e.type === ExtensionType.User))
+			.then(local => {
+				return this.tipsService.getVisualizerExtensions().then((recommmended) => {
+					const installedExtensions = local.map(x => `${x.publisher}.${x.name}`);
+					return this.extensionsWorkbenchService.queryGallery(token).then((pager) => {
+						// filter out installed extensions and the extensions not in the recommended list
+						pager.firstPage = pager.firstPage.filter((p) => {
+							const extensionId = `${p.publisher}.${p.name}`;
+							return installedExtensions.indexOf(extensionId) === -1 && recommmended.findIndex(ext => ext.extensionId === extensionId) !== -1;
+						});
+						pager.total = pager.firstPage.length;
+						pager.pageSize = pager.firstPage.length;
+						return this.getPagedModel(pager);
+					});
+				});
+			});
+	}
+	// End of {{SQL CARBON EDIT}}
```

Comment:
please reformat to `// {{SQL CARBON EDIT}} - ...` the beginning of the tag should be consistent to help with searching the source